### ISSUE TITLE
libyaml-perl/1.25-1 breaks rex/1.6.0-1 test suite (FTBFS and autopkgtest failure)

### ISSUE THREAD

**Issue by gregoa** [RexOps/Rex#1197]
We have the following bug reported to the Debian package of Rex
(https://bugs.debian.org/898578):

It doesn't seem to be a bug in the packaging, so you may want to take
a look. Thanks!

------8<-----------8<-----------8<-----------8<-----------8<-----

```
Source: rex
Version: 1.6.0-1
Severity: serious
Control: affects -1 src:libyaml-perl
User: debian-ci@lists.debian.org
Usertags: needs-update
Tags: upstream buster sid
Justification: fails to build from source

----- Forwarded message from Paul Gevers <elbrus@debian.org> -----

From: Paul Gevers <elbrus@debian.org>
To: libyaml-perl@packages.debian.org, rex@packages.debian.org
Subject: libyaml-perl/1.25-1 breaks rex/1.6.0-1 autopkgtest in testing
Date: Sun, 13 May 2018 19:29:44 +0100

Dear maintainers,

[This e-mail is automatically sent. V3.1 (20180513)]

tl;dr: libyaml-perl/1.25-1 breaks rex/1.6.0-1 autopkgtest in testing
see: https://ci.debian.net/packages/r/rex/testing/amd64/

â¦

----- End forwarded message -----

This also causes a build failure in rex:

YAML Error: Invalid element in map
   Code: YAML_LOAD_ERR_BAD_MAP_ELEMENT
   Line: 6
   Document: 1
 at /usr/share/perl5/YAML/Loader.pm line 361.
t/report.t ....................... 
1..3
ok 1 - 'created report class' isa 'Rex::Report::Base'
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 2/3 subtests 
â¦
Test Summary Report
-------------------
t/report.t                     (Wstat: 65280 Tests: 1 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 3 tests but ran 1.
Files=70, Tests=1519, 30 wallclock secs ( 0.33 usr  0.09 sys + 14.64 cusr  3.44 csys = 18.50 CPU)
Result: FAIL
Failed 1/70 test programs. 0/1519 subtests failed.

When run manually:

t/report.t .. 
1..3
ok 1 - 'created report class' isa 'Rex::Report::Base'
not ok 2 - a new file was created.

ok 3 - the file was not changed
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/3 subtests 

Test Summary Report
-------------------
t/report.t (Wstat: 256 Tests: 3 Failed: 1)
  Failed test:  2
  Non-zero exit status: 1
Files=1, Tests=3,  1 wallclock secs ( 0.02 usr  0.01 sys +  0.26 cusr  0.05 csys =  0.34 CPU)
Result: FAIL



Cheers,
gregor

```
------8<-----------8<-----------8<-----------8<-----------8<-----

Thanks for considering,
  gregor herrmann,
  Debian Perl Group

**Comment by gregoa** [comment]
According to @ntyni's debugging in the abve mentioned bug report, this looks like a regression in yaml-pm

**Comment by perlpunk** [comment]
Should be fixed, could you test with YAML.pm 1.25_002?

**Comment by ntyni** [comment]
On Thu, May 17, 2018 at 06:03:17AM -0700, Tina Müller (tinita) wrote:
> Should be fixed, could you test with YAML.pm 1.25_002?

Many thanks for the quick fix! Yes, I can confirm that rex now passes
its test suite again here with 1.25_002.
-- 
Niko

**Comment by ferki** [comment]
I can confirm that `t/report.t` is broken with YAML-1.25 and 1.25_001, but works with the fix in 1.25_002.

I plan to take a look about how to exclude YAML-1.25 from accepted prerequisites, and release it as part of the bugfixes queueing up for a Rex-1.6.1 release.

By the way, amazing collaboration on this one, folks! Your efforts are much appreciated, well done!

**Comment by perlpunk** [comment]
@ferki I think I want to release 1.26 soon. I'm quite sure there are no new bugs introduced with it ;-)
